src/Console/Commands/MakeLaradiumResource.php

```diff
@@ -40,68 +45,140 @@ public function __construct()
      */
     public function handle()
     {
-        $name = $this->argument('name');
-        $translations = $this->option('t');
-        $api = $this->option('api');
-        $model = $this->option('m');
-        $namespace = str_replace('\\', '', app()->getNamespace());
-
-        $url = strtolower(preg_replace('/(?<!^)[A-Z]/', '-$0', $name));
+        $this->name = $this->argument('name');
 
-        $resourceDirectory = app_path('Laradium/Resources');
+        $resourceDirectory = $this->getResourceDirectory();
         if (!file_exists($resourceDirectory)) {
             File::makeDirectory($resourceDirectory, 0755, true);
             $this->info('Creating resources directory');
         }
 
-        // Resource
-        $dummyResource = File::get(__DIR__ . '/../../../stubs/laradium-resource.stub');
-        $resource = str_replace('{{namespace}}', $namespace, $dummyResource);
-        $resource = str_replace('{{resource}}', $name, $resource);
-        $resource = str_replace('{{resource}}', $name, $resource);
-        $resource = str_replace('{{modelNamespace}}', config('laradium.default_models_directory', 'App'), $resource);
-        $resourceFilePath = app_path('Laradium/Resources/' . $name . 'Resource.php');
+        $resource = $this->prepareResource();
+        $resourceFilePath = $resourceDirectory . '/' . $this->name . 'Resource.php';
 
+//         Create the resource
         if (!file_exists($resourceFilePath)) {
             File::put($resourceFilePath, $resource);
         }
 
-        // API Resource
-        if ($api) {
-            Artisan::call('laradium:api-resource', [
-                'name' => $name
-            ]);
+        $this->createApiResource();
+
+        $this->createModel();
+
+        if (!$this->option('shared')) {
+            $this->createMenuItem();
         }
 
-        if ($model) {
-            Artisan::call('make:model', [
-                'name'        => 'Models/' . $name,
-                '--migration' => true
-            ]);
+        $this->info('Resource successfully created!');
+
+        return;
+    }
+
+    /**
+     * @return string
+     */
+    private function getResourceStub(): string
+    {
+        if ($this->option('shared')) {
+            return File::get(__DIR__ . '/../../../stubs/laradium-shared-resource.stub');
+        }
+
+        return File::get(__DIR__ . '/../../../stubs/laradium-resource.stub');
+    }
+
+    /**
+     * @return string
+     */
+    private function prepareResource(): string
+    {
+        $namespace = $this->getNamespace();
+
+        $resourceStub = $this->getResourceStub();
+        $resource = str_replace('{{namespace}}', $namespace, $resourceStub);
```

Comment:
You can use arrays here, that way you won't have to call `str_replace` multiple times

example:
`str_replace(['{url}', '{hash}'], [asset('wheel/js/wheeltrick.js'), $wheel->hash], $content)`